src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/CSharpDiagnosticWorkerWithAnalyzers.cs

```diff
@@ -235,12 +235,16 @@ private async Task AnalyzeDocument(Project project, ImmutableArray<DiagnosticAna
                 }
                 else if (allAnalyzers.Any()) // Analyzers cannot be called with empty analyzer list.
                 {
-                    var semanticDiagnosticsWithAnalyzers = await compiled
-                        .WithAnalyzers(allAnalyzers, workspaceAnalyzerOptions)
+                    var compilationWithAnalyzers = compiled.WithAnalyzers(allAnalyzers, new CompilationWithAnalyzersOptions(workspaceAnalyzerOptions,
+                        onAnalyzerException: OnAnalyzerException,
+                        concurrentAnalysis: false,
```

Comment:
actually, this is the reason why it works now.
when we use the old code `CompilationWithAnalyzersOptions` gets initialized to the default values and by default it sets `concurrentAnalysis: true`. Looks like then something deadlocks internally and it never completes because of that.

Ultimately I'd like us to be able to recognize why this is happening and how we can avoid that problem but short term this is OK to do it this way as with the current set up it doesn't work at all.

Reply:
actually I found an answer myself http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/AnalyzerHelper.cs,365
looks like VS also switches off concurrency

```
                // in IDE, we always set concurrentAnalysis == false otherwise, we can get into thread starvation due to
                // async being used with syncronous blocking concurrency.
```